lib/qr_code_scanner/test/scanner_test.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2022 Sharezone UG (haftungsbeschränkt)
+// Licensed under the EUPL-1.2-or-later.
+//
+// You may obtain a copy of the Licence at:
+// https://joinup.ec.europa.eu/software/page/eupl
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+
+import 'fake_mobile_scanner_controller.dart';
+
+void main() {
+  group('Scanner', () {
+    late FakeMobileScannerController controller;
+
+    setUp(() {
+      controller = FakeMobileScannerController();
+    });
+
+    testWidgets('should turn on the torch when pressing the torch icon',
+        (tester) async {
+      await tester.pumpWidget(
+        MaterialApp(
+          home: Scaffold(
+            body: Scanner(controller: controller),
+          ),
+        ),
+      );
+
+      await tester.tap(find.byKey(const Key('torch-button-widget-test')));
+
+      expect(controller.hasTorch, false);
```

Comment:
Is the torch on by default or why is it `expect(controller.hasTorch, false); `? Even in this case the test title would be wrong. I don't feel like I understand this test.

Reply:
Yes, reading it a second time the test makes no sense 😂